### PR TITLE
SAKE: Support file uploads from Mario Kart Wii

### PR DESCRIPTION
**Pull request by MikeIsAStar**
Helps address issue #43.

### CODE REVIEW

File: sake/mario_kart_wii.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+	return contentType[startIndex:]
+}
+
+func isRKGDFileValid(rkgdFile []byte) bool {
+	const rkgdFileMagic = 0x524B4744 // RKGD
+
+	if binary.BigEndian.Uint32(rkgdFile[:4]) != rkgdFileMagic {
+		return false
+	}
```
Comment (noahpistilli) [inline sake/mario_kart_wii.go:186]: You could use `bytes.Equal` here.
```go 
rkgdFileMagic := []byte{'R', 'K', 'G', 'D'}

if !bytes.Equal(rkgdFile[:4], rkgdFileMagic) {
    return false
}
```

File: race/nintendo_racing_service.go [hunk 2]
```diff
@@ -66,78 +98,103 @@ func handleNintendoRacingServiceRequest(moduleName string, responseWriter http.R
...
+		logging.Error(moduleName, "Failed to get the Top 10 rankings")
+		writeErrorResponse(raceServiceResultDatabaseError, responseWriter)
+		return
 	}
 
-	responseData := []rankingsResponseData{
-		{
+	numberOfRankings := len(topTenRankings)
+	data := make([]rankingsResponseData, 0, numberOfRankings)
+	for i := 0; i < numberOfRankings; i++ {
```
[the remark this answers is not in the document]
Comment (MikeIsAStar) [reply race/nintendo_racing_service.go]: Addressed in commit [ff8144f](https://github.com/WiiLink24/wfc-server/pull/63/commits/ff8144f89f33683efa7ef7a8e5955574db3452af).

File: sake/mario_kart_wii.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+	courseIdInt, err := strconv.Atoi(courseIdString)
+	if err != nil {
+		logging.Error(moduleName, "Invalid course id")
+		responseWriter.Header().Set(common.SakeFileResultHeader, strconv.Itoa(common.SakeFileResultMissingParameter))
+		return
+	}
+	courseId := common.MarioKartWiiCourseId(courseIdInt)
+	if courseId < common.MarioCircuit {
```
[the remark this answers is not in the document]
Comment (MikeIsAStar) [reply sake/mario_kart_wii.go]: Addressed in commit [ff8144f](https://github.com/WiiLink24/wfc-server/pull/63/commits/ff8144f89f33683efa7ef7a8e5955574db3452af).

File: sake/mario_kart_wii.go (hunk 1 above)
Comment (MikeIsAStar) [reply sake/mario_kart_wii.go]: Addressed in commit [ff8144f](https://github.com/WiiLink24/wfc-server/pull/63/commits/ff8144f89f33683efa7ef7a8e5955574db3452af).

**Pull request merged**